### PR TITLE
Adding ping and sending mail after ping operation's

### PR DESCRIPTION
**Pull request by raskin-aleksandr**
Need to be updated to work with list

### CODE REVIEW

**Comment by orius123** [comment]
@meirwah didn't we decided onm a `properties` folder?

**Comment by oritstone** [comment]
@meirwah please review & merge. Pay attention to Eliya's comment above

**Comment by meirwah** [comment]
@orius123 it is in a properties folder... of the base...

**Comment by meirwah** [comment]
merge issues .. please update @raskin-aleksandr

File: org/openscore/slang/base/files/copy.sl [hunk 1]
```diff
@@ -31,17 +31,19 @@ operation:
         try:
```
Comment (meirwah) [inline org/openscore/slang/base/files/copy.sl]: please separate fixes you make to a different Pull request...
This is very important, otherwise we need to review many changes un-related one to the other...  (and there is more reasons )

File: org/openscore/slang/base/network/ping.sl [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          else:
+            response = os.system("ping -n 1 -i " + ttl + ' -l ' + size + ' -w ' + timeout + ' ' + address)
+          if response == 0:
+            message = (address, 'is up!')
+          else:
+            message = (address, 'is down!')
+          print (message)
+          result = True
+        except Exception as e:
+          message = sys.exc_info()[0]
```
Comment (meirwah) [inline org/openscore/slang/base/network/ping.sl]: didn't we say  sys.exc_info()[0] is not helpful?

File: org/openscore/slang/base/network/resource_health_check.sl [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+# 
+####################################################
+namespace: org.openscore.slang.base.network
+
+imports:
+  network: org.openscore.slang.base.network
+  mail: org.openscore.slang.base.mail
+
+flow:
+  name: resource_health_check
```
Comment (meirwah) [inline org/openscore/slang/base/network/resource_health_check.sl]: name here (resource_health_check) is not good..
maybe ping_hosts ?

@SamMarkowitz WDYT?

Comment (SamMarkowitz) [inline org/openscore/slang/base/network/resource_health_check.sl]: ping_hosts sounds fine to me

**Comment by meirwah** [comment]
@raskin-aleksandr please update the branch...

File: content/org/openscore/slang/base/network/example/ping_hosts.sl [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+          do:
+            network.ping:
+              - address
+        publish:
+              - message_body: "fromInputs['message_body'].append(message)"
+        navigate:
+          SUCCESS: mail_send
+          FAILURE: failure_mail_send
+
+    - mail_send:
```
Comment (meirwah) [inline content/org/openscore/slang/base/network/example/ping_hosts.sl]: does it mean that after mail_send step you will also go to failure_mail_send step??

Comment (meirwah) [inline content/org/openscore/slang/base/network/example/ping_hosts.sl]: you need to use the "- on_failure:" notion...

File: content/org/openscore/slang/base/network/example/ping_hosts.sl [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+####################################################
+# This flow pings addressed from input list and send an email with results
+#
+# Inputs:
+# - address - list of ip's to be checked
+#
+# For correct work of email part - system_property (mail.yaml) file should be filled and called as '--spf' input
+#
+# Results:
+#  SUCCESS: addressee will get an email with result
```
Comment (orius123) [inline content/org/openscore/slang/base/network/example/ping_hosts.sl]: use `-` instead of `:`

File: content/org/openscore/slang/base/network/ping.sl [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+namespace: org.openscore.slang.base.network
+
+operation:
+  name: ping
+
+  inputs:
+    - address
+    - ttl:
+        default: "'64'"
+        required: false
```
Comment (orius123) [inline content/org/openscore/slang/base/network/ping.sl]: @meirwah do we need required if there is a default?

File: content/org/openscore/slang/base/network/ping.sl [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        try:
+          import os, smtplib
+          if os.path.isdir('/usr'):
+            response = os.system("ping -c 1 -t " + ttl + ' -s ' + size + ' -W ' + timeout + ' ' + address)
+          else:
+            response = os.system("ping -n 1 -i " + ttl + ' -l ' + size + ' -w ' + timeout + ' ' + address)
+          if response == 0:
+            message = (address, 'is up!')
+          else:
+            message = (address, 'is down!')
```
Comment (orius123) [inline content/org/openscore/slang/base/network/ping.sl]: isn't it a failure also?
